### DISCUSSION TITLE
Decrease Image Size of all Images in Note

### DISCUSSION THREAD

**Discussion by chrisgrieser** [SilentVoid13/Templater#336] [Templates Showcase]
Per default, Obsidian does display the images with their maximum size. However, often you do not want the image in full size, for example when its a small image with high resolution (which is the case for screenshots on Macs with Retina Display).

While you can [resize images with Obsidian](https://help.obsidian.md/How+to/Embed+files#Resize+images), it's a bit tedious to to this every time you insert an image. 

This templater template automates this task by inserting ` | 400 ` to all images in the current note. All images that already have some resizing, are ignored. the value "400" can easily be changed at the top of the script. 

```js
<%*
//CONFIGURATION: set your desired image size here
const image_px = 400;

//---------------

//get file content and insert image resizing via regex
var noteContent = tp.file.content.replace (
   /^ *(!\[\[[^\|]*?)\]\]$/gm,
   "$1 | " + image_px.toString() + " ]]"
);

//select all in note
let cmEditorAct = this.app.workspace.activeLeaf.view.editor;
cmEditorAct.setSelection({ line: 0, ch: 0 }, { line: 9999, ch: 9999 });

//return regexed content + set cursor at the start
cursor = tp.file.cursor(0);
tR = cursor + noteContent;
%>
```

**Comment by tyf2018** [comment]
Very useful, thanks

**Comment by tyf2018** [comment]
Hi， @chrisgrieser  
After obsidian is upgraded to 0.13.19, an error will be reported after execution. Is it a compatibility issue?

**Reply by chrisgrieser** [reply]
no idea – I don't the the script anymore, sorry :(

maybe ask in the #templater Discord channel?
